### PR TITLE
[BugFix] Fix memo phase still has logical project operator bug

### LINKED ISSUE #16984: [sqlancer] Unknown error
### Steps to reproduce the behavior (Required)
```sql
DROP DATABASE IF EXISTS w10;
CREATE DATABASE w10;
USE w10;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 DATE NOT NULL ,c_0_1 DECIMAL(37, 8)  NOT NULL ,c_0_2 DATE NOT NULL ,c_0_3 DATETIME NULL ,c_0_4 DATE NOT NULL ,c_0_5 DATETIME NOT NULL ,c_0_6 DECIMAL(13, 0)  NULL ,c_0_7 BIGINT NULL ,c_0_8 BIGINT MAX NOT NULL ,c_0_9 DATETIME MAX NOT NULL ,c_0_10 HLL HLL_UNION NULL ,c_0_11 DECIMAL(26, 8)  MAX NULL ,c_0_12 DOUBLE REPLACE NULL ,c_0_13 SMALLINT MAX NULL ,c_0_14 CHAR(31) REPLACE_IF_NOT_NULL NULL ,c_0_15 SMALLINT REPLACE NOT NULL ,c_0_16 DOUBLE REPLACE_IF_NOT_NULL NULL ,c_0_17 BIGINT REPLACE NOT NULL ) AGGREGATE KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4,c_0_5,c_0_6,c_0_7) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 1186 day)) DISTRIBUTED BY HASH (c_0_1,c_0_4,c_0_5,c_0_2,c_0_3,c_0_6,c_0_7) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 DATE NOT NULL,c_1_1 VARCHAR(31) NOT NULL,c_1_2 SMALLINT NOT NULL,c_1_3 INT NOT NULL,c_1_4 VARCHAR(1) NOT NULL,c_1_5 BIGINT NOT NULL,c_1_6 DATETIME NOT NULL,c_1_7 BOOLEAN NOT NULL,c_1_8 BOOLEAN NOT NULL,c_1_9 BOOLEAN NOT NULL,c_1_10 DATETIME NOT NULL,c_1_11 BOOLEAN NOT NULL,c_1_12 VARCHAR(1) NOT NULL,c_1_13 ARRAY<STRING> NOT NULL,c_1_14 ARRAY<STRING> NOT NULL, INDEX index1_c_1_0(`c_1_0`) USING BITMAP) PRIMARY KEY (c_1_0,c_1_1,c_1_2,c_1_3,c_1_4,c_1_5,c_1_6,c_1_7,c_1_8,c_1_9,c_1_10,c_1_11,c_1_12) PARTITION BY RANGE(c_1_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 593 day)) DISTRIBUTED BY HASH (c_1_12,c_1_1,c_1_0,c_1_8,c_1_2,c_1_11,c_1_9,c_1_4,c_1_10,c_1_3,c_1_6,c_1_5) BUCKETS 3 properties("replication_num"="3", "bloom_filter_columns"="c_1_2, c_1_5") ;
CREATE TABLE IF NOT EXISTS t2 (c_2_0 DATE NULL ,c_2_1 DECIMAL(10, 0)  NULL ,c_2_2 DECIMAL(12, 4)  NOT NULL ,c_2_3 DECIMAL(1, 0)  NOT NULL ,c_2_4 ARRAY<STRING> NOT NULL ,c_2_5 VARCHAR(11) NULL ,c_2_6 ARRAY<INT> NOT NULL , INDEX index1_c_2_0(`c_2_0`) USING BITMAP) UNIQUE KEY (c_2_0,c_2_1,c_2_2,c_2_3) PARTITION BY RANGE(c_2_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 105 day)) DISTRIBUTED BY HASH (c_2_0) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t0 VALUES ('2010-01-04', 403100605314765108445.61822466, '2010-01-19', '2010-01-07 08:00:14', '2010-01-14', '2010-01-10 08:00:14', 9175465853287, NULL, -2147055719, '2010-01-28 08:00:14', hll_hash('Px'), NULL, -2.147055719E9, 2, 'QslkDOky', 10, 0.9345251935673141, -1037257802);
INSERT INTO t1 VALUES ('2010-01-30', '', 10, 9, '', -1809327796, '2010-01-03 08:00:14', false, false, true, '2009-12-31 08:00:14', true, '', ["Iht", "t", "9qncvuVy", "tA", "I", "T38jeQMBQ", "c", "o", "br", "7D"], ["bVD5W", "cmga", "sGaajgTT", "", "om", "Rfw3tFRkp", "g", "M", "yJc"]);
DELETE FROM t2 WHERE c_2_2=1758.2428;
CREATE VIEW v0 AS SELECT t0_117.c_0_0, t0_117.c_0_1, t0_117.c_0_2, t0_117.c_0_3, t0_117.c_0_4, t0_117.c_0_5, t0_117.c_0_6, t0_117.c_0_7, t0_117.c_0_8, t0_117.c_0_9, t0_117.c_0_11, t0_117.c_0_12, t0_117.c_0_13, t0_117.c_0_14, t0_117.c_0_15, t0_117.c_0_16, t0_117.c_0_17 FROM t0 AS t0_117 WHERE true GROUP BY t0_117.c_0_0, t0_117.c_0_1, t0_117.c_0_2, t0_117.c_0_3, t0_117.c_0_4, t0_117.c_0_5, t0_117.c_0_6, t0_117.c_0_7, t0_117.c_0_8, t0_117.c_0_9, t0_117.c_0_11, t0_117.c_0_12, t0_117.c_0_13, t0_117.c_0_14, t0_117.c_0_15, t0_117.c_0_16, t0_117.c_0_17 ORDER BY t0_117.c_0_7, t0_117.c_0_2, t0_117.c_0_13, t0_117.c_0_3, t0_117.c_0_16, t0_117.c_0_8, t0_117.c_0_6, t0_117.c_0_2, t0_117.c_0_4, t0_117.c_0_1, t0_117.c_0_16, t0_117.c_0_6, t0_117.c_0_1, t0_117.c_0_11, t0_117.c_0_8, t0_117.c_0_7, t0_117.c_0_5;
TRUNCATE TABLE t2;
SELECT t2_126.c_2_0 FROM t1 AS t1_125, t0 AS t0_124,  t2 AS t2_126 INNER JOIN (SELECT v0_127.c_0_0, v0_127.c_0_1, v0_127.c_0_2, v0_127.c_0_3, v0_127.c_0_4, v0_127.c_0_5, v0_127.c_0_6, v0_127.c_0_7, v0_127.c_0_8, v0_127.c_0_9, v0_127.c_0_11, v0_127.c_0_12, v0_127.c_0_13, v0_127.c_0_14, v0_127.c_0_15, v0_127.c_0_16, v0_127.c_0_17 FROM v0 AS v0_127 WHERE EXISTS (SELECT v0_127.c_0_7 FROM t1 AS t1_125, v0 AS v0_127 WHERE true) ) subv0 ON t2_126.c_2_5 = subv0.c_0_2 AND t2_126.c_2_5 > subv0.c_0_2 AND t2_126.c_2_1 != subv0.c_0_1 AND t2_126.c_2_1 = subv0.c_0_1;
```

fe.log
```Prolog
java.lang.IllegalStateException: null
	at com.google.common.base.Preconditions.checkState(Preconditions.java:494) ~[spark-dpp-1.0.0.jar:?]
	at com.starrocks.sql.optimizer.statistics.Statistics.getColumnStatistic(Statistics.java:56) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.statistics.ExpressionStatisticCalculator$ExpressionStatisticVisitor.visitVariableReference(ExpressionStatisticCalculator.java:63) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.statistics.ExpressionStatisticCalculator$ExpressionStatisticVisitor.visitVariableReference(ExpressionStatisticCalculator.java:42) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator.accept(ColumnRefOperator.java:110) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.statistics.ExpressionStatisticCalculator.calculate(ExpressionStatisticCalculator.java:39) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.statistics.ExpressionStatisticCalculator.calculate(ExpressionStatisticCalculator.java:35) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.statistics.PredicateStatisticsCalculator$PredicateStatisticsCalculatingVisitor.getExpressionStatistic(PredicateStatisticsCalculator.java:337) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.statistics.PredicateStatisticsCalculator$PredicateStatisticsCalculatingVisitor.visitBinaryPredicate(PredicateStatisticsCalculator.java:216) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.statistics.PredicateStatisticsCalculator$PredicateStatisticsCalculatingVisitor.visitBinaryPredicate(PredicateStatisticsCalculator.java:51) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.operator.scalar.BinaryPredicateOperator.accept(BinaryPredicateOperator.java:61) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.statistics.PredicateStatisticsCalculator.statisticsCalculate(PredicateStatisticsCalculator.java:34) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.statistics.StatisticsCalculator.estimateStatistics(StatisticsCalculator.java:1255) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.statistics.StatisticsCalculator.computeJoinNode(StatisticsCalculator.java:815) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.statistics.StatisticsCalculator.visitLogicalJoin(StatisticsCalculator.java:694) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.statistics.StatisticsCalculator.visitLogicalJoin(StatisticsCalculator.java:138) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.operator.logical.LogicalJoinOperator.accept(LogicalJoinOperator.java:149) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.statistics.StatisticsCalculator.estimatorStats(StatisticsCalculator.java:154) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.rule.join.JoinOrder.calculateStatistics(JoinOrder.java:229) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.rule.join.JoinReorderLeftDeep.enumerate(JoinReorderLeftDeep.java:79) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.rule.join.JoinOrder.reorder(JoinOrder.java:179) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.rule.join.ReorderJoinRule.enumerate(ReorderJoinRule.java:75) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.rule.join.ReorderJoinRule.transform(ReorderJoinRule.java:152) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.Optimizer.memoOptimize(Optimizer.java:352) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.Optimizer.optimizeByCost(Optimizer.java:150) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.optimizer.Optimizer.optimize(Optimizer.java:92) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.StatementPlanner.createQueryPlan(StatementPlanner.java:95) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:66) ~[starrocks-fe.jar:?]
	at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:37) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:373) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:313) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:430) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:676) ~[starrocks-fe.jar:?]
	at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:55) ~[starrocks-fe.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
```
### Real behavior (Required)
```sql
mysql> SELECT t2_126.c_2_0 FROM t1 AS t1_125, t0 AS t0_124,  t2 AS t2_126 INNER JOIN (SELECT v0_127.c_0_0, v0_127.c_0_1, v0_127.c_0_2, v0_127.c_0_3, v0_127.c_0_4, v0_127.c_0_5, v0_127.c_0_6, v0_127.c_0_7, v0_127.c_0_8, v0_127.c_0_9, v0_127.c_0_11, v0_127.c_0_12, v0_127.c_0_13, v0_127.c_0_14, v0_127.c_0_15, v0_127.c_0_16, v0_127.c_0_17 FROM v0 AS v0_127 WHERE EXISTS (SELECT v0_127.c_0_7 FROM t1 AS t1_125, v0 AS v0_127 WHERE true) ) subv0 ON t2_126.c_2_5 = subv0.c_0_2 AND t2_126.c_2_5 > subv0.c_0_2 AND t2_126.c_2_1 != subv0.c_0_1 AND t2_126.c_2_1 = subv0.c_0_1;
ERROR 1064 (HY000): Unknown error
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```sql
mysql> select current_version();
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-2.5-RELEASE c1d1113 |
+----------------------------+
1 row in set (0.01 sec)
```

### PR DESCRIPTION
**Pull request by kangkaisen**
Signed-off-by: kangkaisen <kangkaisen@apache.org>

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes  https://github.com/StarRocks/starrocks/issues/16984

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

for https://github.com/StarRocks/starrocks/issues/16984

The directly cause is when JoinAssociateBaseRule, we set projection to a logical project operator, and the projection lost.
```
        if (CollectionUtils.isNotEmpty(splitter.getBotJoinChildCols())) {
            RowOutputInfo mergedRow = newBotJoinLeftChild.getRowOutputInfo().addColsToRow(
                    splitter.getBotJoinChildCols(),
                    newBotJoinLeftChild.getOp().getProjection() != null);
            Operator.Builder builder = OperatorBuilderFactory.build(newBotJoinLeftChild.getOp());

            Operator newBotJoinLeftChildOp = builder.withOperator(newBotJoinLeftChild.getOp())
                    .setProjection(new Projection(mergedRow.getColumnRefMap())).build();
```
```
        @Override
        public LogicalProjectOperator build() {
            return new LogicalProjectOperator(columnRefMap, this.limit);
        }
```

The root cause is after MergeProjectWithChildRule, we shouldn't have LogicalProjectOperator. but `RemoveAggregationFromAggTable` generate LogicalProjectOperator.

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoveAggregationFromAggTable.java [hunk 1]
```diff
@@ -126,9 +127,9 @@ public boolean check(OptExpression input, OptimizerContext context) {
         }
         List<String> groupKeyColumns = aggregationOperator.getGroupingKeys().stream()
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoveAggregationFromAggTable.java:128]: can directly  `Collectors.toSet()` in this stream

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/JoinReorderProperty.java [hunk 2]
```diff
@@ -81,7 +81,7 @@ private JoinReorderProperty() {
         LEFT_ASSCOM_PROPERTY[JoinOperator.LEFT_OUTER_JOIN.ordinal()] = new int[] {1, 1, 1, 1, 0, 0, 0, 2, 0, 0};
         LEFT_ASSCOM_PROPERTY[JoinOperator.LEFT_SEMI_JOIN.ordinal()] = new int[] {0, 1, 1, 1, 0, 0, 0, 0, 0, 0};
         LEFT_ASSCOM_PROPERTY[JoinOperator.LEFT_ANTI_JOIN.ordinal()] = new int[] {0, 1, 1, 1, 0, 0, 0, 0, 0, 0};
-        LEFT_ASSCOM_PROPERTY[JoinOperator.FULL_OUTER_JOIN.ordinal()] = new int[] {1, 2, 0, 0, 0, 0, 0, 2, 0, 0};
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/JoinReorderProperty.java:84]: Other pr has changed. can revert this change.

**Review by packy92 — DISMISSED** [review]

**Review by packy92 — DISMISSED** [review]

**Review by packy92 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 8 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalProjectOperator.java|0|2|00.00%|[120, 121]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/RemoveAggregationFromAggTable.java|4|4|100.00%|[]|

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by ABingHuang — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**